File: jofafrazze-csharp/day01/Day01.cs
using AdventOfCode;
using System;

namespace aoc
{
    public class Day01
    {
        static readonly string day = "day01";

        // Sonar Sweep: Count increments

        public static Object PartA(string file)
        {
            var input = ReadInput.Ints(day, file);
            int ans = 0;
            for (int i = 1; i < input.Count; i++)
                if (input[i] > input[i - 1])
                    ans++;
            return ans;
        }

        public static Object PartB(string file)
        {
            var v = ReadInput.Ints(day, file);
            int ans = 0;
            for (int i = 1; i < v.Count - 2; i++)
                if (v[i] + v[i + 1] + v[i + 2] > v[i - 1] + v[i] + v[i + 1])
                    ans++;
            return ans;
        }

        static void Main() => Aoc.Execute(day, PartA, PartB);
    }
}

File: saidaspen-kotlin/src/Day01.kt
import java.io.File

fun main() {
    println(Day01.part1())
    println(Day01.part2())
}

object Day01 {
    private var input = File("/home/mendlock/.aoc/202101").readText()
    fun part1() = ints(input).windowed(2).count { it[1] > it[0] }
    fun part2() = ints(input).windowed(4).count { it[3] > it[0] }
}